test/models/drive_test.dart

```diff
@@ -0,0 +1,1003 @@
+      final Ride ride = RideFactory().generateFake(status: RideStatus.approved);
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.pending),
+          RideFactory().generateFake(status: RideStatus.cancelledByDriver),
+          RideFactory().generateFake(status: RideStatus.pending),
+          ride,
+          RideFactory().generateFake(status: RideStatus.rejected),
+          RideFactory().generateFake(status: RideStatus.cancelledByRider),
+          RideFactory().generateFake(status: RideStatus.withdrawnByRider),
+        ],
+      );
+      expect(drive.approvedRides, [ride]);
+    });
+    test('return correct rides with rides have more than 1 approved ones in it', () async {
+      final Ride ride1 = RideFactory().generateFake(status: RideStatus.approved);
+      final Ride ride2 = RideFactory().generateFake(status: RideStatus.approved);
+      final Ride ride3 = RideFactory().generateFake(status: RideStatus.approved);
+
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.pending),
+          RideFactory().generateFake(status: RideStatus.cancelledByDriver),
+          RideFactory().generateFake(status: RideStatus.pending),
+          ride1,
+          ride2,
+          RideFactory().generateFake(status: RideStatus.rejected),
+          RideFactory().generateFake(status: RideStatus.cancelledByRider),
+          RideFactory().generateFake(status: RideStatus.withdrawnByRider),
+          ride3,
+        ],
+      );
+      expect(drive.approvedRides, [ride1, ride2, ride3]);
+    });
+  });
+  group('Drive.pendingRides', () {
+    test('return null if rides is null', () async {
+      final Drive drive = DriveFactory().generateFake(
+        createDependencies: false,
+      );
+      expect(drive.pendingRides, null);
+    });
+    test('return an empty List if rides has no pending rides', () async {
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.approved),
+        ],
+      );
+      expect(drive.pendingRides, []);
+    });
+    test('return correct ride with rides having one pending ride', () async {
+      final Ride ride = RideFactory().generateFake(status: RideStatus.pending);
+      final Drive drive = DriveFactory().generateFake(
+        rides: [
+          RideFactory().generateFake(status: RideStatus.approved),
+          RideFactory().generateFake(status: RideStatus.cancelledByDriver),
+          ride,
+          RideFactory().generateFake(status: RideStatus.rejected),
+          RideFactory().generateFake(status: RideStatus.cancelledByRider),
+          RideFactory().generateFake(status: RideStatus.withdrawnByRider),
+        ],
+      );
+      expect(drive.pendingRides, [ride]);
+    });
+    test('return correct rides with rides have more than 1 pending ride', () async {
```

Comment:
> Please keep the test cases minimal. They should document the code and be readable. Every test case should be specific. Here, for example, there is no need for one Ride of every other status to be in the list, since it that has already been tested in the test above.

Here, too